src/data/set/intervals/basic.lean

```diff
@@ -272,6 +272,16 @@ lemma Icc_subset_Icc_iff (h₁ : a₁ ≤ b₁) :
 ⟨λ h, ⟨(h ⟨le_refl _, h₁⟩).1, (h ⟨h₁, le_refl _⟩).2⟩,
  λ ⟨h, h'⟩ x ⟨hx, hx'⟩, ⟨le_trans h hx, le_trans hx' h'⟩⟩
 
+lemma Icc_ssubset_Icc_left (hI : a₂ ≤ b₂) (ha : a₂ < a₁) (hb : b₁ ≤ b₂) :
+  Icc a₁ b₁ ⊂ Icc a₂ b₂ :=
+(ssubset_iff_of_subset (Icc_subset_Icc (le_of_lt ha) hb)).mpr
+  ⟨a₂, left_mem_Icc.mpr hI, not_and.mpr (λ f g, lt_irrefl a₂ (lt_of_lt_of_le ha f))⟩
+
+lemma Icc_ssubset_Icc_right (hI : a₂ ≤ b₂) (ha : a₂ ≤ a₁) (hb : b₁ < b₂) :
+  Icc a₁ b₁ ⊂ Icc a₂ b₂ :=
+(ssubset_iff_of_subset (Icc_subset_Icc ha (le_of_lt hb))).mpr
+  ⟨b₂, right_mem_Icc.mpr hI, (λ f, lt_irrefl b₁ (lt_of_lt_of_le hb f.2))⟩
+
```

Comment:
I think it probably makes sense to put these below or above all the `_iff` lemmas, rather than cutting them into two pieces.